lib/msal-common/src/authority/Authority.ts

```diff
@@ -435,4 +453,36 @@ export class Authority {
     isAlias(host: string): boolean {
         return this.metadata.aliases.indexOf(host) > -1;
     }
+
+    /**
+     * Rebuild the authority string with the region
+     * 
+     * @param host string
+     * @param region string 
+     */
+    static buildRegionalAuthorityString(host: string, region: string): string {
+        // Create and validate a Url string object with the initial authority string
+        const globalUrl = new UrlString(host);
+        globalUrl.validateAsUri();
+
+        // Include the query string portion of the url
+        return UrlString.constructAuthorityUriFromObject({
+            ...globalUrl.getUrlComponents(),
+            HostNameAndPort: `${region}.login.microsoft.com`
```

Comment:
Did we test this for all regions? This pattern had some issues before if I recall with `/token` end point I assumed. As long as we are dynamically reading the `token endpoint` and not constructing, we should be fine though.